### ISSUE TITLE
Advice for training multiple faces during one session?

### ISSUE THREAD

**Issue by brandonkoch3** [TheLastBen/fast-stable-diffusion#1196]
I'm curious if there is specified guidance when wanting to train multiple faces during one session.  For example, lets say I want to train my face and my friend's face.  I have a folder of images labeled "bob (1).jpg, bob (2).jpg, etc." and a folder of images labeled "sally (1).jpg, sally (2).jpg, etc."  I've tried a few different approaches, but I'm never able to really find an output that works for me.  In the end, my goal prompt might be "A photo of Bob and Sally standing in front of the Eiffel Tower."

In my attempts, I've tried;
- Uploading all images of Bob and Sally simultaneously, and ending up with 15 images of Bob and 15 images of Sally.  I then modify the training steps to around 1,950 steps, and the text encoder training steps to around 750.  When I test the trained model, and try "A photo of Bob," or just "Bob" in the prompt, I end up with a person who looks like a blend of Bob and Sally.
- Uploading images of Bob and training with roughly 975 steps and 375 text encoder training steps.  Upon completion, re-training that same .ckpt, this time with photos of Sally, and again training with roughly 975 steps and 375 text encoder steps.  When I test the trained model, it seems to predominately generate images of Sally, even when using "Bob" in the prompt.

Is there advice for getting the best output with two or more people?

**Comment by hartmamt** [comment]
Yeah, I have the same experience. Works like a dream for photos of a person and a pet, but I haven't gotten anything good when trying to prompt for two people that I've trained the model on.

**Comment by StanPHC** [comment]
Based on how diffusion models work, you will probably need to try and try and try in order to get generated images containing both your instances. It's simply to difficult to assess while going through the inference steps to what extend the image is converging to the prompt (most likely it'll start by generating a default face and slowly filling in the details, trying to fit both characteristics from bob as from sally into the face). 

Instead, you could generate seperate images of Bob and Sally from your model trained on both instances. You can then use the same model for in-/outpainting to make it one picture. Alternatively you can generate two people near the Eiffeltower with your model and then try to inpaint Bob and Sally.
